src/commands/api-mesh/__tests__/log-get.test.js

```diff
@@ -0,0 +1,186 @@
+			'rayId': 'ray1',
+			'URL': 'http://example.com',
+			'Request Method': 'GET',
+			'Response Status': 200,
+		});
+	});
+
+	afterEach(() => {
+		platformSpy.mockRestore();
+	});
+
+	test('snapshot FetchLogsCommand', () => {
+		expect(FetchLogsCommand.description).toMatchInlineSnapshot(
+			`"Get the Log of a given mesh by RayId"`,
+		);
+		expect(FetchLogsCommand.args).toMatchInlineSnapshot(`
+		        [
+		          {
+		            "description": "to fetch the log ",
+		            "name": "rayId",
+		            "required": true,
+		          },
+		        ]
+	    `);
+		expect(FetchLogsCommand.flags).toMatchInlineSnapshot(`
+		                {
+		                  "ignoreCache": {
+		                    "allowNo": false,
+		                    "char": "i",
+		                    "default": false,
+		                    "description": "Ignore cache and force manual org -> project -> workspace selection",
+		                    "parse": [Function],
+		                    "type": "boolean",
+		                  },
+		                }
+	        `);
+		expect(FetchLogsCommand.aliases).toMatchInlineSnapshot(`[]`);
+	});
+});
+
+test('should handle log not found error', async () => {
+	getLogsByRayId.mockRejectedValue(new Error('LogNotFound'));
+
+	const runResult = FetchLogsCommand.run();
+
+	return runResult.catch(err => {
+		expect(err.message).toMatchInlineSnapshot(`
+		"No logs found for RayID ray1. Check the RayID and try again. RequestId: dummy_request_id. Alternatively, you can use the following command to get all logs for a 30 minute time period: 
+		aio api-mesh log-get-bulk --startTime YYYY-MM-DDTHH:MM:SSZ --endTime YYYY-MM-DDTHH:MM:SSZ --filename mesh_logs.csv"
+	`);
+		expect(logSpy.mock.calls).toMatchInlineSnapshot(`[]`);
+		expect(errorLogSpy.mock.calls).toMatchInlineSnapshot(`
+		[
+		  [
+		    "No logs found for RayID ray1. Check the RayID and try again. RequestId: dummy_request_id. Alternatively, you can use the following command to get all logs for a 30 minute time period: 
+		aio api-mesh log-get-bulk --startTime YYYY-MM-DDTHH:MM:SSZ --endTime YYYY-MM-DDTHH:MM:SSZ --filename mesh_logs.csv",
+		  ],
+		]
+	`);
+	});
+});
+
+test('should handle server error', async () => {
+	getLogsByRayId.mockRejectedValue(new Error('Internal Server Error. Please try again later.'));
+
+	const runResult = FetchLogsCommand.run();
+
+	return runResult.catch(err => {
+		expect(err.message).toMatchInlineSnapshot(
+			`"Unable to get mesh logs. Please check the details and try again. If the error persists please contact support. RequestId: dummy_request_id"`,
+		);
+		expect(logSpy.mock.calls).toMatchInlineSnapshot(`[]`);
+		expect(errorLogSpy.mock.calls).toMatchInlineSnapshot(`
+			[
+			  [
+			    "Unable to get mesh logs. Please check the details and try again. If the error persists please contact support. RequestId: dummy_request_id",
+			  ],
+			]
+		`);
+	});
+});
+
+test('should handle generic error', async () => {
```

Comment:
What condition would trigger the generic error "Unable to get logs"? I understand we handle both 404 and 500 errors, but I’m curious about what other scenarios might result in this error message.

Reply:
I was able to check this condition when  TI delegator has timed out or isnt runnning, but i think  it can also happen if the server responds with an unexpected HTTP status code that is not specifically handled or if there is an error in the code that is not specifically handled. 

my intuition for naming it as a generic error message was because it serves as a fallback for any unhandled errors, and essentially its the command which has failed entirely. but yeah can be stated other way as well 

<img width="1060" alt="image" src="https://github.com/user-attachments/assets/87c68d62-c348-4b46-a79a-fc39a8f1919f">
Case - ServerError happens when SMS has failed respond
<img width="887" alt="image" src="https://github.com/user-attachments/assets/d607c24d-16b0-468b-bbe0-d97a9e5e2fb5">